front/lib/api/assistant/agent.ts

```diff
@@ -33,15 +36,45 @@ export async function* runAgent(
   | AgentMessageSuccessEvent,
   void
 > {
-  // TODO(multi-actions): Implement isLegacyAgent and fork execution here.
-
-  for await (const event of runLegacyAgent(
+  const fullConfiguration = await getAgentConfiguration(
```

Comment:
Let's pass it to runLegacyAgent to avoid double query here.

Reply:
I'm doing just that no ?